### ISSUE TITLE
Database queue drive does not dispatch event JobQueued for each jobs when use Bus::batch

### ISSUE THREAD

**Issue by RuslanMelnychenko** [laravel/framework#52380]
### Laravel Version

10.43.0

### PHP Version

8.2

### Database Driver & Version

MySQL

### Description

When I dispatch Bus::batch with jobs which using database connection, database driver dont dispatch JobQueued and JobQueueing for each job.
[In the code](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Queue/DatabaseQueue.php#L150), database driver use mass insert to table, without dispatch events. For example Redis driver has [another flow for dispatch job](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Queue/RedisQueue.php#L109).

### Steps To Reproduce

Add listener of `Illuminate\Queue\Events\JobQueued` then dispatch jobs by:
```php
Bus::batch($jobs)->onConnection('database')->dispatch()
```
and you see listener does not trigger for dispatched jobs.

**Comment by driesvints** [comment]
Hey there,

Can you first please try one of the support channels below? If you can actually identify this as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb) or [InertiaJS Discord](https://discord.gg/inertiajs) for Inertia questions
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!

**Comment by RuslanMelnychenko** [comment]
This is a bug because the behavior when using different queue drivers is different, which breaks the consistency of the system in different environments. I have made sure that all drivers except database send Illuminate\Queue\Events\JobQueued when sending Bus::batch($jobs)->dispatch(). 
Some Laravel add-ons use this event to extend the job functionality. What I encountered with the problem when using https://github.com/mateusjunges/trackable-jobs-for-laravel with `->onConnection('database')`
